dev/src/index.ts

```diff
@@ -597,6 +597,23 @@ export class Firestore implements firestore.Firestore {
           // also set the `protocol` option for GAX fallback to force http
           if (useFallback) {
             settings.protocol = 'http';
+
+            // If the emulator mode is enabled, we set emulator credentials
+            // to prevent the gRPC-fallback client from trying to obtain
+            // service account credentials from the environment.
+            if (process.env.FIRESTORE_EMULATOR_HOST) {
+              const emulatorCredentials = {
+                access_token: 'owner',
+              };
+              const emulatorClient =
+                // eslint-disable-next-line node/no-extraneous-require
+                new (require('google-auth-library').OAuth2Client)();
```

Comment:
Since `google-auth-library` is not a direct dependency (it is a transient dependency of `google-gax`) We get `extraneous-require` lint error here. I think disabling it for this line is okay... WDYT?

Reply:
If we added 'google-auth-library' to nodejs-firestore dependencies, would that also resolve the issue? Or am I missing some other context?

Reply:
Yep, adding `google-auth-library` to `package.json` would resolve this, but I am not sure if that is necessary here as `google-gax` should pull in `google-auth-library` as a dependency anyway.

Reference: https://github.com/mysticatea/eslint-plugin-node/blob/master/docs/rules/no-extraneous-require.md

> If a `require()`'s target is extraneous (it's not written in `package.json`), the program works in local, but will not work after dependencies are re-installed. It will cause troubles to your team/contributors. This rule disallows `require()` of extraneous modules.